### PR TITLE
Add BitCore BTX support

### PR DESCRIPTION
**Pull request by dalijolijo**

### CODE REVIEW

**Comment by NicolasDorier** [comment]
You need first to integrate with NBitcoin and NBXplorer and be sure that tests pass for your altcoin.

File: BTCPayServer/Properties/launchSettings.json [hunk 1]
```diff
@@ -7,12 +7,13 @@
                 "BTCPAY_NETWORK": "regtest",
                 "BTCPAY_BUNDLEJSCSS": "true",
                 "BTCPAY_LTCEXPLORERURL": "http://127.0.0.1:32838/",
+                "BTCPAY_BTXEXPLORERURL": "http://127.0.0.1:32838/",
                 "BTCPAY_BTCLIGHTNING": "type=charge;server=http://127.0.0.1:54938/;api-token=foiewnccewuify",
                 "BTCPAY_BTCEXTERNALLNDGRPC": "type=lnd-grpc;server=https://lnd:lnd@127.0.0.1:53280/;allowinsecure=true",
                 "BTCPAY_BTCEXTERNALLNDREST": "type=lnd-rest;server=https://lnd:lnd@127.0.0.1:53280/lnd-rest/btc/;allowinsecure=true;macaroonfilepath=D:\\admin.macaroon",
                 "BTCPAY_BTCEXPLORERURL": "http://127.0.0.1:32838/",
                 "ASPNETCORE_ENVIRONMENT": "Development",
-                "BTCPAY_CHAINS": "btc,ltc",
```
Comment (NicolasDorier) [inline BTCPayServer/Properties/launchSettings.json]: don't change the launchSettings

File: BTCPayServer.Tests/docker-compose.yml [hunk 2]
```diff
@@ -77,7 +79,7 @@ services:
       - "32838"
     environment:
       NBXPLORER_NETWORK: regtest
-      NBXPLORER_CHAINS: "btc,ltc"
```
Comment (NicolasDorier) [inline BTCPayServer.Tests/docker-compose.yml]: don't change the docker-compose for tests

File: BTCPayServer.Tests/UnitTest1.cs [hunk 3]
```diff
@@ -1069,6 +1069,70 @@ public void CanHaveLTCOnlyStore()
             }
         }
```
Comment (NicolasDorier) [inline BTCPayServer.Tests/UnitTest1.cs]: don't add a test for your coin.

**Comment by NicolasDorier** [comment]
Remove everything but InitBitcore

File: BTCPayServer/Properties/launchSettings.json (hunk 1 above)
Comment (dalijolijo) [reply BTCPayServer/Properties/launchSettings.json]: Ok. I followed the LTC integration. Is this not correct?

**Comment by dalijolijo** [comment]
I removed everything except InitBitcore.

**Comment by dalijolijo** [comment]
@NicolasDorier It seems, that BTX integration into  NBXplorer not yet available because the [Add BTX support PR #126](https://github.com/dgarage/NBXplorer/pull/126) is not part of current NBXplorer.Client Version 1.0.3.4 (see [here](https://github.com/dgarage/NBXplorer/compare/Client/v1.0.3.4...master))

Please create a new NBXplorer.Client Version and update [BTCPayServer/BTCPayServer.csproj](https://github.com/btcpayserver/btcpayserver/blob/1d7081d8b8d2e831cb5841fef8d29f834a944742/BTCPayServer/BTCPayServer.csproj#L52)
    `<PackageReference Include="NBXplorer.Client" Version="1.0.3.4" />`

**Comment by dalijolijo** [comment]
NBXplorer.Client Version [1.0.3.5](https://github.com/dgarage/NBXplorer/commit/da7df8601938e5af732cb1fab577fa5d994539c4) is available. 
@NicolasDorier Please update [BTCPayServer/BTCPayServer.csproj](https://github.com/btcpayserver/btcpayserver/blob/1d7081d8b8d2e831cb5841fef8d29f834a944742/BTCPayServer/BTCPayServer.csproj#L52) to NBXplorer.Client Version 1.0.3.5
    `<PackageReference Include="NBXplorer.Client" Version="1.0.3.**4**" />`

**Comment by dalijolijo** [comment]
#378

**Comment by dalijolijo** [comment]
@NicolasDorier ready to be merged

**Comment by NicolasDorier** [comment]
Bitcore support removed because of cryptopedia being hacked https://github.com/btcpayserver/btcpayserver/commit/884acdde322429a46390c5a76001f1b37efa5a5c and breaking btcpay tests

**Comment by NicolasDorier** [comment]
Uncomment and see. The test that check that the default rate provider work. (which it does not because cryptopedia is down)

**Comment by dalijolijo** [comment]
See https://github.com/btcpayserver/btcpayserver/pull/535

**Pull request merged**